frontend/index.js

```diff
@@ -65,32 +100,77 @@ function UrlPreviewBlock() {
     });
 
     const base = useBase();
-    const table = base.getTableByIdIfExists(cursor.activeTableId);
+    const activeTable = base.getTableByIdIfExists(cursor.activeTableId);
 
-    // table is briefly null when switching to a newly created table.
-    if (!table) {
+    // activeTable is briefly null when switching to a newly created activeTable.
+    if (!activeTable) {
         return null;
     }
 
     return (
-        <RecordPreview
-            table={table}
-            selectedRecordId={selectedRecordId}
-            selectedFieldId={selectedFieldId}
-        />
+        <Box display="flex">
+            {isSettingsVisible ? (
+                <SettingsForm
+                    settings={settingsValidationResult.settings}
+                    setIsSettingsVisible={setIsSettingsVisible}
+                />
+            ) : (
+                <RecordPreview
+                    activeTable={activeTable}
+                    settingsValidationResult={settingsValidationResult}
+                    selectedRecordId={selectedRecordId}
+                    selectedFieldId={selectedFieldId}
+                />
+            )}
+        </Box>
     );
 }
 
 // Shows a preview, or a message about what the user should do to see a preview.
-function RecordPreview({table, selectedRecordId, selectedFieldId}) {
+function RecordPreview({activeTable, settingsValidationResult, selectedRecordId, selectedFieldId}) {
+    const {settings, isValid, message} = settingsValidationResult;
+    const {isEnforced, urlField} = settings;
     const [isDialogOpen, setIsDialogOpen] = useState(false);
+    let table = activeTable;
+    let content;
+
+    if (!isValid) {
+        content = (
+            <Container>
+                <Text>{message}</Text>
+            </Container>
+        );
+    }
+
+    // If the creator has specified a Table and Field for URL previews...
+    if (isEnforced && settings.table) {
+        table = settings.table;
```

Comment:
I think this may cause a bug. When we set `table` to `settings.table`, the subsequent call to `table.getFieldByIdIfExists` will always fail if the user is viewing a different table. That will cause `selectedField` to be null, and that will cause `selectedRecord` to be null, and that will cause the `content` which states, "Switch to the [...] table" to be overwritten with a more generic message.